finagle-http/src/main/scala/com/twitter/finagle/http/filter/DtabFilter.scala

```diff
@@ -40,7 +40,41 @@ object DtabFilter {
     Future.value(rsp)
   }
 
+  @deprecated("use DtabFilter.Decoder", "2016-06-07")
   class Finagle[Req <: Message] extends DtabFilter[Req, Response] {
     def respondToInvalid(req: Req, msg: String) = invalidResponse(msg)
   }
+
+  /**
+   * Extracts Dtab-local headers from incoming requests and adds the
+   * Dtab to the local context.
+   */
+  class Extractor extends DtabFilter[Request, Response] {
+    def respondToInvalid(req: Request, msg: String) = invalidResponse(msg)
+  }
+
+  /**
+   * Modifies each request with Dtab encoding from Dtab.local and
+   * streams chunked responses via `Reader`.  If the request already
+   * contains Dtab headers they will be dropped silently.
+   */
+  class Injector extends SimpleFilter[Request, Response] {
+    def apply(req: Request, service: Service[Request, Response]): Future[Response] = {
+      // XXX won't this behave poorly with retries?
```

Comment:
The fact that Dtab headers were silently removed from finagle requests has cost many hours of debugging, I would prefer that we keep it if it's not too hairy.